skimage/registration/_tvl1.py

```diff
@@ -0,0 +1,206 @@
+        maintain attachement and regularization parts in
+        correspondence.
+    nwarp : int
+        Number of times I1 is warped.
+    niter : int
+        Number of fixed point iteration.
+    tol : float
+        Tolerance used as stopping criterion based on the L² distance
+        between two consecutive values of (u, v).
+    prefilter : bool
+        whether to prefilter the estimated optical flow before each
+        image warp.
+
+    Returns
+    -------
+    flow : ~numpy.ndarray
+        The estimated optical flow.
+
+    """
+
+    grid = np.meshgrid(*[np.arange(n) for n in I0.shape], indexing='ij')
+
+    f0 = lambda_ * tau
+    f1 = dt / tau
+    tol *= I0.size
+
+    flow = flow0
+
+    g = np.zeros((I0.ndim, ) + I0.shape)
+    proj = np.zeros((I0.ndim, I0.ndim, ) + I0.shape)
+
+    s_g = [slice(None), ] * g.ndim
+    s_p = [slice(None), ] * proj.ndim
+    s_d = [slice(None), ] * (proj.ndim-2)
+
+    for _ in range(nwarp):
+        if prefilter:
+            flow = ndi.filters.median_filter(flow, [1]+I0.ndim*[3])
+
+        wI1 = warp(I1, grid+flow, mode='nearest')
+        grad = np.array(np.gradient(wI1))
+        NI = (grad*grad).sum(0)
+        NI[NI == 0] = 1
+
+        rho_0 = wI1 - I0 - (grad*flow0).sum(0)
+
+        for _ in range(niter):
+
+            # Data term
+
+            rho = rho_0 + (grad*flow).sum(0)
+
+            idx = abs(rho) <= f0 * NI
+
+            flow_ = flow
+
+            flow_[:, idx] -= rho[idx]*grad[:, idx]/NI[idx]
+
+            idx = ~idx
+            srho = f0 * np.sign(rho[idx])
+            flow_[:, idx] -= srho*grad[:, idx]
+
+            # Regularization term
+            flow = flow_.copy()
+
+            for idx in range(flow.shape[0]):
+                s_p[0] = idx
+                for _ in range(2):
+                    for ax in range(flow.shape[0]):
+                        s_g[0] = ax
+                        s_g[ax+1] = slice(0, -1)
+                        g[tuple(s_g)] = np.diff(flow[idx], axis=ax)
+                        s_g[ax+1] = slice(None)
+
+                    norm = np.sqrt((g ** 2).sum(0))[np.newaxis, ...]
+                    norm *= f1
+                    norm += 1.
+                    proj[idx] -= dt * g
+                    proj[idx] /= norm
+
+                    # d will be the (negative) divergence of p[idx]
+                    d = -proj[idx].sum(0)
+                    for ax in range(flow.shape[0]):
+                        s_p[1] = ax
+                        s_p[ax+2] = slice(0, -1)
+                        s_d[ax] = slice(1, None)
+                        d[tuple(s_d)] += proj[tuple(s_p)]
+                        s_p[ax+2] = slice(None)
+                        s_d[ax] = slice(None)
+
+                    flow[idx] = flow_[idx] + d
+
+        flow0 -= flow
+        if (flow0*flow0).sum() < tol:
+            break
+
+        flow0 = flow
+
+    return flow
+
+
+def tvl1(I0, I1, dt=0.2, lambda_=15, tau=0.3, nwarp=5, niter=10,
```

Comment:
Since we are in Python 3.5+, use `def optical_flow(image0, image1, *, dt=0.2, ...)` to indicate that arguments after image1 *must* be given as keywords.

Reply:
Could this function be named more relevantly as well? Something like `optical_flow_tvl1`, maybe.

Reply:
I think `optical_flow` by itself is enough. Optical flow is the name of what we are computing. TV-L1 is an implementation detail. If we add other methods in the future, we can add a `method=` keyword argument.

Reply:
I would vote for `optical_flow_tvl1` since if we implement an other algorithm for optical flow, the function's parameters will be different.